src/analysis/convex/cone/proper.lean

```diff
@@ -0,0 +1,46 @@
+/-
+Copyright (c) 2022 Apurva Nakade All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Apurva Nakade
+-/
+import analysis.inner_product_space.adjoint
+
+/-!
+
+We define the closure of the convex cone over a real inner product space as a convex cone.
+
+-/
+
+-- TODO: add proper cones and farkas' lemma
+
+
+open continuous_linear_map filter
+
+namespace convex_cone
+
+variables {E : Type*} [inner_product_space ℝ E]
+
+/-- The closure of a convex cone inside a real inner product space is a convex cone. This
+construction is mainly used for defining maps between proper cones. -/
+def closure (K : convex_cone ℝ E) : convex_cone ℝ E :=
+{ carrier := closure ↑K,
+  smul_mem' :=
+  begin
+    rw ← seq_closure_eq_closure,
+    rintros c hc x ⟨seq, mem, tends⟩,
+    exact ⟨λ n, c • seq n, ⟨λ n, K.smul_mem hc (mem n), tendsto.const_smul tends c⟩⟩,
+  end,
+  add_mem' :=
+  begin
+    rw ← seq_closure_eq_closure,
+    rintros x ⟨xseq, xmem, xtends⟩ y ⟨yseq, ymem, ytends⟩,
+    exact ⟨λ n, xseq n + yseq n, ⟨λ n, K.add_mem (xmem n) (ymem n), tendsto.add xtends ytends⟩⟩,
+  end }
```

Comment:
Can you use [set.maps_to.closure_left](https://leanprover-community.github.io/mathlib_docs/topology/basic.html#set.maps_to.closure_left) and [map_mem_closure₂](https://leanprover-community.github.io/mathlib_docs/topology/constructions.html#map_mem_closure%E2%82%82)?

Reply:
I'll try it out. This sequential version seemed more natural to me from a mathematical perspective. But l'll see if the functions you suggest provide better proofs. Thanks!